plugins/bigquery/dbt/include/bigquery/macros/materializations/incremental.sql

```diff
@@ -28,14 +29,37 @@
       {% set build_sql = create_table_as(False, target_relation, sql) %}
   {% else %}
      {% set dest_columns = adapter.get_columns_in_relation(existing_relation) %}
+     
+     {%- set dest_partition = none -%}
+     
+     {#-- if partitioned, get the range of partition values to be updated --#}
+     {% if partition_by %}
+         {%- call statement('pro_tmp', fetch_result = True) -%}
+            {{ create_table_as(True, tmp_relation, sql) }}
+         {%- endcall -%}
+
+         {% call statement('get_partition_range', fetch_result = True) %}
```

Comment:
can we use a set block + `run_query` here? We're slowly phasing out instances of `call statement`